### ISSUE TITLE
bypass client-side names through WPE

### ISSUE THREAD

**Issue by Runico** [TrinityCore/TrinityCore#15357]
Information:
http://www.deathsoft.com/forum/index.php/topic/9163-bypass-client-side-names-basic-wpe-tutorial/

Rev: f2aa95c3503fe936f08d9fcc3c7f76f8bfcd8b36
TDB: 59

**Comment by DDuarte** [comment]
>  I know this was fixed on MaNGOS a long time ago and Ascent even longer ago.

**Comment by Runico** [comment]
@DDuarte, It's already works in current TC

**Comment by jackpoz** [comment]
please post the how to reproduce steps in this github issue

**Comment by Runico** [comment]
@jackpoz, Really i don't know how to reproduce, but today i have seen players as name God, Bitch,Fuck, then i google'd it and found that link attachmented above

**Comment by StormBytePP** [comment]
That can be done easily by UPDATE in database of the character's name

**Comment by tkrokli** [comment]
@Runico : Do you know for a fact that those players with "illegal" names do not have access to the DB or are friends with the DB admin and that the only way they could have got those names are via hacks? Have you tested any of those hacks yourself?

**Comment by Runico** [comment]
@tkrokli, I'm the owner and nobody else have access to DB except myself, I'll talk with them how they did that

**Comment by MrSmite** [comment]
Seems like this *might* be exploitable in Trinity based on the tutorial. Assuming these are the DBC involved in name restrictions:

> DeclinedWord.dbc
 DeclinedWordCases.dbc
 NamesProfanity.dbc
 NamesReserved.dbc

it doesn't appear that (in the 335 branch) *CheckPlayerName()* ever checks any DBC. When it is called by *HandleCharCreateOpcode()* it seems to only check length and special characters.

__Note:__ Of course, I haven't actually tried it ;)

**Comment by ariel-** [comment]
I have successfully tested this, but instead of using WPE I created a MPQ file with those 2 files empty. Cientside, I don't get the "Name is already taken" error, and server accepts it.

DeclinedNames is actually used for name declension (methinks is for russian client), the DBCs involved are NamesProfanity and NamesReserved.

In TC, the table "reserved_name" exists, which allows for blocking certain names serverside, the problem is those DBCs use regexps for matching multiple forbidden names at once, and the table simply does a 1:1 match
